sdk/storage/azure-storage-file-share/azure/storage/fileshare/_download.py

```diff
@@ -181,48 +189,46 @@ def __next__(self):
 
     next = __next__  # Python 2 compatibility.
 
-    def _get_chunk_data(self):
+    def _get_chunk_data(self) -> bytes:
         chunk_data = self._current_content[: self._chunk_size]
         self._current_content = self._current_content[self._chunk_size:]
         return chunk_data
 
 
 class StorageStreamDownloader(object):  # pylint: disable=too-many-instance-attributes
-    """A streaming object to download from Azure Storage.
-
-    :ivar str name:
-        The name of the file being downloaded.
-    :ivar: str path:
-        The full path of the file.
-    :ivar str share:
-        The name of the share where the file is.
-    :ivar ~azure.storage.fileshare.FileProperties properties:
-        The properties of the file being downloaded. If only a range of the data is being
-        downloaded, this will be reflected in the properties.
-    :ivar int size:
-        The size of the total data in the stream. This will be the byte range if specified,
-        otherwise the total size of the file.
-    """
+    """A streaming object to download from Azure Storage."""
+
+    name: str
+    """The name of the file being downloaded."""
+    path: str
+    """The full path of the file."""
+    share: str
+    """The name of the share where the file is."""
+    properties: "FileProperties"
+    """The properties of the file being downloaded. If only a range of the data is being
+        downloaded, this will be reflected in the properties."""
+    size: int
+    """The size of the total data in the stream. This will be the byte range if specified,
+        otherwise the total size of the file."""
 
     def __init__(
-        self,
-        client=None,
-        config=None,
-        start_range=None,
-        end_range=None,
-        validate_content=None,
-        max_concurrency=1,
-        name=None,
-        path=None,
-        share=None,
-        encoding=None,
-        **kwargs
-    ):
+        self, client: "FileOperations" = None,  # type: ignore [assignment]
+        config: "StorageConfiguration" = None,  # type: ignore [assignment]
+        start_range: Optional[int] = None,
+        end_range: Optional[int] = None,
+        validate_content: bool = None,  # type: ignore [assignment]
+        max_concurrency: int = 1,
+        name: str = None,  # type: ignore [assignment]
+        path: str = None,  # type: ignore [assignment]
+        share: str = None,  # type: ignore [assignment]
+        encoding: Optional[str] = None,
+        **kwargs: Any
+    ) -> None:
         self.name = name
         self.path = path
         self.share = share
-        self.properties = None
-        self.size = None
+        self.properties = None  # type: ignore [assignment]
+        self.size = None  # type: ignore [assignment]
```

Comment:
Let's declare `properties` as `Optional` and remove the ignore and start `self.size = 0` to avoid the other ignore.

Reply:
I made the change for `self.size = 0`.

It looks like `properties` is set after the `_initial_request()` call. In `_download_async.py`, this call happens in `_setup()`. So to avoid any confusion, let's get rid of the `self.properties` initialization line, and just set `self.properties` after the `_initial_request()` call, just like blob.